src/main/java/edu/ucsb/cs56/ucsb_courses_search/model/WeeklyView.java

```diff
@@ -0,0 +1,214 @@
+    public int col;
+    public int row;
+    public Cell(int col, int row) {
+      this.col = col;
+      this.row = row;
+    }
+    public boolean equals(Cell other){
+      return (col == other.col) && (row == other.row);
+    }
+  }
+
+  private int col;
+  private int row;
+  private int rowspan;
+  private Iterable<ScheduleItem> myclasses;
+  private ArrayList<Cell> ignoredCells;
+
+  public WeeklyView(Iterable<ScheduleItem> myclasses) {
+    this.col = 0;
+    this.row = 2;
+    this.rowspan = 1;
+    this.myclasses = myclasses;
+    ignoredCells = new ArrayList<Cell>();
+  }
+
+  public String[] getTimeRange() {
+    String[] timerange = new String[30];
+    int index = 0;
+    for (int h = 8; h <= 22; h++) {
+      for (int m = 0; m <= 1; m++) {
+        String time = "";
+        if (h > 12) {
+          if(h - 12 < 10)
+            time += "0";
+          time += (h - 12);
+        } else {
+          if(h < 10)
+            time += "0";
+          time += h;
+        }
+        time += ":";
+        if (m == 0) {
+          time += "00";
+        } else {
+          time += "30";
+        }
+        if (h >= 12) {
+          time += " PM";
+        } else {
+          time += " AM";
+        }
+        timerange[index] = time;
+        index++;
+      }
+    }
+    return timerange;
+  }
+
+  public boolean isColEqualsMeetday(ScheduleItem course, int column) {
+    String time = course.getMeetday();
+    if (column == 0)
+      return time.contains("M");
+    if (column == 1)
+      return time.contains("T");
+    if (column == 2)
+      return time.contains("W");
+    if (column == 3)
+      return time.contains("R");
+    if (column == 4)
+      return time.contains("F");
+    return false;
+  }
+
+  public int getCourseStartRow(ScheduleItem course) {
+    // TESTED
+    // RETURNS ThE ROW, where 8:00AM = 2, 8:30AM = 3
+
+    String times = course.getMeettime();
+    // String times = "11:00 - 11:50";
+    int hour = Integer.parseInt(times.substring(0, 2));
+
+    int minutes = Integer.parseInt(times.substring(3, 5));
+    int time = hour * 60 + minutes;// time in minutes
+    if (time == 480) {
+      return 2;
+    }
+    int row = (time - 480) / 30 + 2;
+    return row;
+  }
+
+  public int getCourseRowSpan(ScheduleItem course) {
+    // Course course, String LectureOrSection
+    // for rowspan nte that meetime gives the section time not lecture time in form
+    // 09:00-09:50
+    // for a 75 minute class, usually 12:30-1:45 or 2-3:14, the hour is difference
+    // of 1
+    // for a 50 min class, the hour is the same etc; 7-7:50
+    // for a long labs, there will be first a 2 hour difference such as 14-16:30 or
+    // 14-16:50, then also check the minutes for an extra
+    String times = course.getMeettime();
+    // String times = "11:00 - 11:50";
+    // if(LectureOrSection=="lecture"){
+    // times = course.getAssociatedLectureTime();
+    // }
+    // else{
+    // times = course.getMeettime();
+    // }
```

Comment:
If this code isn't being used, please remove it, unless you have a reason not to.